apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/BigQueryParameter.Data.cs

```diff
@@ -15,33 +15,124 @@
 using System;
 using System.Data;
 using System.Data.Common;
+using Google.Api.Gax;
 
 namespace Google.Cloud.BigQuery.V2
 {
-    public sealed partial class BigQueryParameter : DbParameter 
+    public sealed partial class BigQueryParameter : DbParameter
+#if NET45 || NET451
+        , ICloneable
+#endif
     {
         /// <inheritdoc />
-        public override void ResetDbType()
+        public override DbType DbType
```

Comment:
Again, maybe dictionaries could simplify this? (Ideally, only specify the mapping in one place and reverse it...) Maybe an internal attribute on BigQueryDbType?

Reply:
it's not 1-1 though.  timestamp maps to datetime, but datetime maps back to  (BQType.)datetime.
for the same reason as above and I felt the switch was more readable by displaying these elided cases better.
let me noodle on it though.